connector/kafka-0-10-sql/src/main/resources/error/kafka-error-conditions.json

```diff
@@ -30,6 +30,13 @@
       "Specified: <specifiedPartitions> Assigned: <assignedPartitions>"
     ]
   },
+  "KAFKA_TIMESTAMP_OFFSET_DOES_NOT_MATCH_ASSIGNED" : {
+    "message" : [
+      "Partitions specified for Kafka timestamp based <position> offsets don't match what are assigned. Maybe topic partitions are created ",
```

Comment:
> Maybe topic partitions are created or deleted while the query is running.

Technically, the starting offset and end offset (for batch) are only valid for the first microbatch (or batch query). That said, there is still a chance of race condition, but it's more likely that users missed to specify some partition(s). The above message rules out the second case which I think is more likely (and original message was meant to be).